r3bdata/califaData/R3BCalifaHitData.h

```diff
@@ -27,73 +27,74 @@ class R3BCalifaHitData : public FairMultiLinkedData
-    R3BCalifaHitData(UInt_t Nb, Double_t ene, Double_t nf, Double_t ns, Double_t theta, Double_t phi, ULong64_t time);
-
-    R3BCalifaHitData(uint64_t time, double theta, double phi, uint32_t clusterId)
-        : fTime(time)
-        , fTheta(theta)
-        , fPhi(phi)
-        , fClusterId(clusterId)
-        , fEnergy(0.)
-        , fNf(0.)
-        , fNs(0)
-        , fNbOfCrystalHits(0)
-    {
-    }
+    R3BCalifaHitData(std::vector<Int_t> crystalList, Double_t ene, Double_t nf, Double_t ns, Double_t theta, Double_t phi, ULong64_t time,Int_t clusterType);
+
+    // R3BCalifaHitData(std::vector<Int_t> crystalList, Double_t ene, Double_t nf, Double_t ns, Double_t theta, Double_t phi, ULong64_t time,Int_t clusterType)
+    //     : fCrystalList(crystalList)
+    //     , fEnergy(ene)
+    //     , fNf(nf)
+    //     , fNs(ns)
+    //     , fTheta(theta)
+    //     , fPhi(phi)
+    //     , fTime(time)
+    //     , fClusterType(clusterType)
+    //
+    // {
+    // }
 
     /** Copy constructor **/
     R3BCalifaHitData(const R3BCalifaHitData&);
 
-    /** += operator **/
-    R3BCalifaHitData& operator+=(R3BCalifaCrystalCalData& cH)
-    {
-        this->fEnergy += cH.GetEnergy();
-        this->fNf += cH.GetNf();
-        this->fNs += cH.GetNs();
-        this->fNbOfCrystalHits++;
-        return *this;
-    }
+    // /** += operator **/
+    // R3BCalifaHitData& operator+=(R3BCalifaCrystalCalData& cH)
+    // {
+    //     this->fEnergy += cH.GetEnergy();
+    //     this->fNf += cH.GetNf();
+    //     this->fNs += cH.GetNs();
+    //     this->fNbOfCrystalHits++;
+    //     return *this;
+    // }
 
     R3BCalifaHitData& operator=(const R3BCalifaHitData&) { return *this; }
 
     /** Destructor **/
     virtual ~R3BCalifaHitData() {}
 
     /** Accessors **/
-    UInt_t GetNbOfCrystalHits() const { return fNbOfCrystalHits; }
     Double_t GetEnergy() const { return fEnergy; }
     Double_t GetNf() const { return fNf; }
     Double_t GetNs() const { return fNs; }
     Double_t GetTheta() const { return fTheta; }
     Double_t GetPhi() const { return fPhi; }
     ULong64_t GetTime() const { return fTime; }
-    uint32_t GetClusterId() const { return fClusterId; }
+    uint32_t GetClusterType() const { return fClusterType; }
+    std::vector<Int_t> GetCrystalList(){return fCrystalList; }
+
+    Int_t GetNbCrystals() const {return fCrystalList.size();}
 
     /** Modifiers **/
-    void SetNbOfCrystalHits(UInt_t number) { fNbOfCrystalHits = number; }
     void SetEnergy(Double_t ene) { fEnergy = ene; }
     void SetNf(Double_t nf) { fNf = nf; }
     void SetNs(Double_t ns) { fNs = ns; }
     void SetTheta(Double_t theta) { fTheta = theta; }
     void SetPhi(Double_t phi) { fPhi = phi; }
     void SetTime(ULong64_t time) { fTime = time; }
-    void SetClusterId(uint32_t id) { fClusterId = id; }
-
+    void SetClusterType(uint32_t id) { fClusterType = id; }
+    void SetCrystalList(std::vector<Int_t> list) {fCrystalList = list;}
```

Comment:
Avoid expensive copy assignment.
```suggestion
    void SetCrystalList(std::vector<Int_t> list) {fCrystalList = std::move(list);}
```